controllers/storagecluster/cephfilesystem.go

```diff
@@ -164,30 +163,48 @@ func (obj *ocsCephFilesystems) ensureCreated(r *StorageClusterReconciler, instan
 }
 
 func (r *StorageClusterReconciler) createDefaultSubvolumeGroup(filesystemName, filesystemNamespace string, ownerReferences []metav1.OwnerReference) error {
-	// TODO: After fix of rook issue https://github.com/rook/rook/issues/13220 add svg name spec
 
 	existingsvg := &cephv1.CephFilesystemSubVolumeGroup{}
-	err := r.Client.Get(r.ctx, types.NamespacedName{Name: defaultSubvolumeGroupName, Namespace: filesystemNamespace}, existingsvg)
+	defaultPinningValue := 1
+	cephsvg := &cephv1.CephFilesystemSubVolumeGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            generateNameForCephSubvolumeGroup(filesystemName),
+			Namespace:       filesystemNamespace,
+			OwnerReferences: ownerReferences,
+		},
+		Spec: cephv1.CephFilesystemSubVolumeGroupSpec{
+			Name:           defaultSubvolumeGroupName,
+			FilesystemName: filesystemName,
+			Pinning: cephv1.CephFilesystemSubVolumeGroupSpecPinning{
+				Distributed: &defaultPinningValue,
+			},
+		},
+	}
+
+	err := r.Client.Get(r.ctx, types.NamespacedName{Name: cephsvg.Name, Namespace: cephsvg.Namespace}, existingsvg)
 	if err == nil {
 		if existingsvg.DeletionTimestamp != nil {
 			r.Log.Info("Unable to restore subvolumegroup because it is marked for deletion.", "subvolumegroup", klog.KRef(filesystemNamespace, defaultSubvolumeGroupName))
-			return fmt.Errorf("failed to restore subvolumegroup %s because it is marked for deletion", existingsvg.Name)
+			return fmt.Errorf("Failed to restore subvolumegroup %s because it is marked for deletion", existingsvg.Name)
 		}
-	}
 
-	objectMeta := metav1.ObjectMeta{Name: defaultSubvolumeGroupName, Namespace: filesystemNamespace, OwnerReferences: ownerReferences}
-	cephFilesystemSubVolumeGroup := &cephv1.CephFilesystemSubVolumeGroup{ObjectMeta: objectMeta}
-	mutateFn := func() error {
-		cephFilesystemSubVolumeGroup.Spec = cephv1.CephFilesystemSubVolumeGroupSpec{
-			FilesystemName: filesystemName,
+		r.Log.Info("Restoring original CephSubvolumeGroup.", "CephSubvolumeGroup", klog.KRef(cephsvg.Namespace, cephsvg.Name))
+		existingsvg.ObjectMeta.OwnerReferences = cephsvg.ObjectMeta.OwnerReferences
+		existingsvg.Spec = cephsvg.Spec
+		err = r.Client.Update(context.TODO(), existingsvg)
+		if err != nil {
+			r.Log.Error(err, "Unable to update CephSubvolumeGroup.", "CephFileSystem", "CephSubvolumeGroup", klog.KRef(cephsvg.Namespace, cephsvg.Name))
+			return err
+		}
+	} else if errors.IsNotFound(err) {
+		r.Log.Info("Creating CephSubvolumeGroup.", "CephSubvolumeGroup", klog.KRef(cephsvg.Namespace, cephsvg.Name))
+		err = r.Client.Create(context.TODO(), cephsvg)
+		if err != nil {
+			r.Log.Error(err, "Unable to create CephSubvolumeGroup.", "CephSubvolumeGroup", klog.KRef(cephsvg.Namespace, cephsvg.Name))
+			return err
 		}
-		return nil
-	}
-	_, err = ctrl.CreateOrUpdate(r.ctx, r.Client, cephFilesystemSubVolumeGroup, mutateFn)
```

Comment:
we should use `create or update` what if user want to update the pinning configuration